### PR TITLE
wallpaper: depends on macos

### PR DESCRIPTION
**Pull request by iMichka**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/wallpaper.rb [hunk 1]
```diff
@@ -14,6 +14,7 @@ class Wallpaper < Formula
   end
 
   depends_on xcode: ["11.4", :build]
+  depends_on :macos
   depends_on macos: :sierra
```
Comment (carlocab) [inline Formula/wallpaper.rb:18]: I feel like there's a lot of redundancy in these three lines.

Comment (iMichka) [reply Formula/wallpaper.rb:18]: Almost; we had this situation elsewhere.

`depends_on macos: :sierra` is a no-op on linux. Same for the Xcode dep.

The goal is to have a clean error message on `brew install` on Linux:

```
wallpaper: macOS is required for this software.
Error: An unsatisfied requirement failed this build.
```

 I could remove the checks around high Sierra and the old Xcode version, as we don't care anymore. Is that OK?

Comment (carlocab) [reply Formula/wallpaper.rb]: I think we still need the Xcode check, as it implicitly rules out Mojave (only has 11.3).

Comment (carlocab) [reply Formula/wallpaper.rb]: Alternatively, we can make it
```ruby
depends_on macos: :catalina
```

Comment (iMichka) [reply Formula/wallpaper.rb]: I went for the Xcode check.

**Review by Bo98 — APPROVED** [review]

**Pull request merged**